bigchaindb/elections/election.py

```diff
@@ -242,24 +244,41 @@ def show_election(self, bigchain):
         return response
 
     @classmethod
-    def approved_update(cls, bigchain, new_height, txns):
-        votes = {}
-        for txn in txns:
-            if not isinstance(txn, Vote):
+    def approved_elections(cls, bigchain, new_height, txns):
+        elections = defaultdict(list)
+        for tx in txns:
+            if not isinstance(tx, Vote):
                 continue
+            election_id = tx.asset['id']
+            elections[election_id].append(tx)
 
-            election_id = txn.asset['id']
-            election_votes = votes.get(election_id, [])
-            election_votes.append(txn)
-            votes[election_id] = election_votes
-
-            election = cls.has_concluded(bigchain, election_id, election_votes, new_height)
-            # Once an election concludes any other conclusion for the same
-            # or any other election is invalidated
-            if election:
-                cls.store_election_results(bigchain, election, new_height)
-                return cls.on_approval(bigchain, election, new_height)
-        return None
+        validator_set_updated = False
+        validator_set_change = []
+        for election_id, votes in elections.items():
+            election = bigchain.get_transaction(election_id)
+
+            if not election.has_concluded(bigchain, election_id, votes, new_height):
+                continue
+
+            if election.makes_validator_set_change():
+                if validator_set_updated:
+                    continue
+                validator_set_change.append(election.get_validator_set_change(bigchain, new_height))
+                validator_set_updated = True
+
+            election.on_approval(bigchain, election, new_height)
+
+        return validator_set_change
+
+    def makes_validator_set_change(self):
```

Comment:
There is no need to wrap the field in a method. We can make it a `@property` in the future if we need to transform it on read.